src/lib.rs

```diff
@@ -264,8 +267,26 @@ impl LightningNode {
         })
     }
 
-    pub fn get_node_info(&self) -> NodeInfo {
-        todo!()
+    pub fn get_node_info(&self) -> Result<NodeInfo> {
+        let node_state: NodeState = self.sdk.node_info().map_err(|e| RuntimeError {
+            code: RuntimeErrorCode::NodeUnavailable,
+            msg: e.to_string(),
+        })?;
+        let rate = self.get_exchange_rate();
+
+        Ok(NodeInfo {
+            node_pubkey: node_state.id,
+            peers: node_state.connected_peers,
+            channels_info: ChannelsInfo {
+                num_channels: 0,
+                num_usable_channels: 0,
```

Comment:
If we keep this here, please add a comment on both lines mentioning the deprecation of the fields, as well as in the UDL.